linker/shared/src/main/scala/org/scalajs/linker/frontend/IRLoader.scala

```diff
@@ -91,20 +104,25 @@ final class IRLoader extends Analyzer.InputProvider with MethodSynthesizer.Input
 private final class ClassDefAndInfoCache {
   private var cacheUsed: Boolean = false
   private var version: Option[String] = None
-  private var cacheUpdate: Future[(ClassDef, Infos.ClassInfo)] = _
+  private var cacheUpdate: Future[(ClassDef, Infos.ClassInfo, List[Infos.TopLevelExportInfo])] = _
 
   def loadInfo(irFile: IRFileImpl)(
       implicit ec: ExecutionContext): Future[Infos.ClassInfo] = {
     update(irFile).map(_._2)
   }
 
+  def loadTopLevelExportInfos(irFile: IRFileImpl)(
+      implicit ec: ExecutionContext): Future[List[Infos.TopLevelExportInfo]] = {
+    update(irFile).map(_._3)
+  }
+
   def loadClassDefAndVersion(irFile: IRFileImpl)(
       implicit ec: ExecutionContext): Future[(ClassDef, Option[String])] = {
     update(irFile).map(s => (s._1, version))
   }
 
   private def update(irFile: IRFileImpl)(
-      implicit ec: ExecutionContext): Future[(ClassDef, Infos.ClassInfo)] = synchronized {
+      implicit ec: ExecutionContext): Future[(ClassDef, Infos.ClassInfo, List[Infos.TopLevelExportInfo])] = synchronized {
```

Comment:
The CI says that this line is too long (more than 120 characters). Perhaps move the `synchronized` inside?

Reply:
I took the chance to simplify this a bit. PTAL